AutoscoperM/AutoscoperM.py

```diff
@@ -1194,3 +1281,62 @@ def convertNodeToData(self, volumeNode: slicer.vtkMRMLVolumeNode) -> vtk.vtkImag
             imageData = imageReslice.GetOutput()
 
         return imageData
+
+    def check4D(self, node: slicer.vtkMRMLNode) -> bool:
+        """
+        Checks if the volume is 4D
+        """
+        self.is_4d = type(node) == slicer.vtkMRMLSequenceNode
```

Comment:
I suggest to revisit this, a static function called `IsSequenceVolume(volumeNode: slicer.vtkMRMLVolumeNode | None)` without storing the state in a logic ivar would be less error-prone